diff -urN arc.org/arcdos.c arc/arcdos.c
--- arc.org/arcdos.c	Wed Apr 15 00:58:21 1992
+++ arc/arcdos.c	Tue Aug 21 19:08:09 2001
@@ -30,15 +30,8 @@
 #if	UNIX
 #include <sys/types.h>
 #include <sys/stat.h>
-#if	BSD
 #include <sys/time.h>
-#else	
 #include <time.h>		/* Sys V. Bleah. */
-struct	timeval {
-	long	tv_sec;
-	long	tv_usec;
-};
-#endif	/* BSD vs SYSV */
 #endif
 
 #if	GEMDOS
diff -urN arc.org/tmclock.c arc/tmclock.c
--- arc.org/tmclock.c	Wed Apr 15 00:58:50 1992
+++ arc/tmclock.c	Tue Aug 21 19:09:27 2001
@@ -14,12 +14,8 @@
 /* Julian day number of the Unix* clock's origin, 01 Jan 1970. */
 #define JD1970 2440587L
 #define	CENTURY	19
-#if	BSD
 #include <sys/time.h>
-#else
 #include <time.h>
-extern long	timezone;	/* should be in <time.h>, but isn't on Sun */
-#endif
 
 long	tzone;
 
--- arc-5.21j/arcdos.c~	2003-10-31 03:32:19.000000000 +0100
+++ arc-5.21j/arcdos.c	2005-09-19 01:15:06.000000000 +0200
@@ -30,17 +30,8 @@
 #if	UNIX
 #include <sys/types.h>
 #include <sys/stat.h>
-#if	BSD
 #include <sys/time.h>
-#else	
 #include <time.h>		/* Sys V. Bleah. */
-#if	NEED_TIMEVAL
-struct	timeval {
-	long	tv_sec;
-	long	tv_usec;
-};
-#endif
-#endif	/* BSD vs SYSV */
 #endif
 
 #if	GEMDOS
